model/src/cfd.rs

```diff
@@ -1361,23 +1382,30 @@ impl AsBlocks for Duration {
     }
 }
 
-// Make a `Margin` newtype and call `Margin::long`
-/// Calculates the long's margin in BTC
-///
-/// The margin is the initial margin and represents the collateral the buyer
-/// has to come up with to satisfy the contract. Here we calculate the initial
-/// long margin as: quantity / (initial_price * leverage)
-pub fn calculate_long_margin(price: Price, quantity: Usd, leverage: Leverage) -> Amount {
-    quantity / (price * leverage)
+/// Detemine the leverage based on role and position
+pub fn long_and_short_leverage(
+    taker_leverage: Leverage,
+    role: Role,
+    position: Position,
+) -> (Leverage, Leverage) {
+    match (role, position) {
+        (Role::Maker, Position::Long) | (Role::Taker, Position::Short) => {
+            (Leverage::ONE, taker_leverage)
+        }
+        (Role::Maker, Position::Short) | (Role::Taker, Position::Long) => {
+            (taker_leverage, Leverage::ONE)
+        }
+    }
 }
 
-/// Calculates the shorts's margin in BTC
+// Make a `Margin` newtype and call `Margin::long`
+/// Calculates the margin in BTC
 ///
-/// The short margin is represented as the quantity of the contract given the
-/// initial price. The short side can currently not leverage the position but
-/// always has to cover the complete quantity.
-pub fn calculate_short_margin(price: Price, quantity: Usd) -> Amount {
-    quantity / price
+/// The margin is the initial margin and represents the collateral both parties have to come up with
```

Comment:
🔧 
```suggestion
/// The initial margin represents the collateral both parties have to come up with
```

"The margin is the initial margin" is a bit redundant imo.